### ISSUE TITLE
Custom comments informix

### ISSUE THREAD

**Issue by npetzall** [schemaspy/schemaspy#425]
Informix doesn't have comments but this could be simulated?

@begooden-it

**Comment by npetzall** [comment]
Yes, there is a selectTableCommentsSql also one for columns that can be added to the databaseType.

It can be injected with schema, and it expects two columns in return.

table_name, comments

So something in the line of
`select table_name, comments from something join something where something = :schema`

**Comment by begooden-it** [comment]
Sorry for the latency.

What do you need exactly? or can I do this myself in some "publicly" (i.e not in source-code) accessible parameters area ?
OR is this something you will need to define within non public aread (source-code)?
I just put a RFE ( request for enhancement) to IBM to implement this, but it may take time in the meantime to have it?

How could we proceed in the meantime?

**Comment by npetzall** [comment]
It's possible to do without any change to the code.

https://schemaspy.readthedocs.io/en/latest/configuration.html#databasetype

https://schemaspy.readthedocs.io/en/latest/configuration.html#other-properties

If it gets implemented we would like to add it to the default. 

You should extend the normal informix databaseType

**Comment by npetzall** [comment]
If you supply the schema for comments, I'll help create the queries and databaseType file.

**Comment by begooden-it** [comment]
Hi,

this  is my custom schema, the RFE will probably take time to implement

--drop table if exists qx_table_comments ;
create table qx_table_comments (
tabid INTEGER,
comments lvarchar(4096)
) ;
create index fk_systables_qx_table_comm on qx_table_comments(tabid) ;
--
--drop table if exists qx_column_comments ;
create table qx_column_comments (
tabid INTEGER,
colno INTEGER,
comments lvarchar(4096)
) ;
create index fk_syscolumns_qx_column_comm on qx_column_comments(tabid,colno) ;

Here is how I insert data for the first time:
insert into qx_table_comments
SELECT t.tabid,"comments on " || t.tabname
from systables t
where t.tabid > 99
and t.tabid not in ( SELECT tabid from qx_table_comments );

insert into qx_column_comments
SELECT c.tabid,c.colno,"comments on " || c.colname
from syscolumns c
where c.tabid > 99
and (c.tabid not in ( SELECT tabid from qx_column_comments ) and c.colno not in ( SELECT colno from qx_column_comments )) ;

and here is how I fetch the data:
For the table comments:
select t.tabname,c.comments
from systables t,qx_table_comments c
where t.tabid = c.tabid ;

for the column comments:
select t.tabname,c.colname,qx_column_comments.comments
from systables t,syscolumns c,qx_column_comments
where t.tabid = qx_column_comments.tabid
and c.tabid = t.tabid
and c.colno = qx_column_comments.colno

**Comment by npetzall** [comment]
So file to same dir as schemaspy-jar named `informix_qx_comments.properties` with the following contents:
```
#
# see http://schemaspy.org/dbtypes.html
# for configuration / customization details
#

# Extended informix database type to support comments as described in https://github.com/schemaspy/schemaspy/issues/425
extends=informix

selectTableCommentsSql=select t.tabname as table_name, c.comments as comments from systables t,qx_table_comments c where t.tabid = c.tabid and t.owner = :schema

selectColumnCommentsSql=select t.tabname as table_name, c.colname as column_name, qx_column_comments.comments as comments from systables t,syscolumns c,qx_column_comments where t.tabid = qx_column_comments.tabid and c.tabid = t.tabid and c.colno = qx_column_comments.colno and t.owner = :schema 
```

Now any custom comments added as per above schemas will end up in the database model and such in the html report.

**Comment by begooden-it** [comment]
Sorry to come back, but: 

which files do I need for the test? I tried adding the informix_qx_comments.properties file and ran the 
 following command at prompt:
java -jar "C:\Tools\SchemaSpy\schemaSpy.jar" -dp "java -jar "C:\Tools\SchemaSpy\schemaSpy.jar" -dp "C:\Program Files\ibm\csdk12.10.FC9\jdbc\lib\ifxjdbc.jar" -t informix_qx_comments.properties -db %1 -host 192.168.1.20:29140 -port 29140 -u informix -p informix -server production_tcp -o %2 -connprops="CLIENT_LOCALE\=en_US.utf8;DB_LOCALE\=en_US.57372" -charset UTF-8 -s informix\ifxjdbc.jar" -t informix -db %1 -host 192.168.1.20:29140 -port 29140 -u informix -p informix -server production_tcp -o %2 -connprops="CLIENT_LOCALE\=en_US.utf8;DB_LOCALE\=en_US.57372" -charset UTF-8 -s informix

But no comments appeared ( and no error). I must be missing something obious for you and not for me :-)

Thanks

**Comment by npetzall** [comment]
@begooden-it I assume some kind of copy paste issue, since that command line is really really strange.

I'll give it ago here
```
java -jar "C:\Tools\SchemaSpy\schemaSpy.jar" -dp "C:\Program Files\ibm\csdk12.10.FC9\jdbc\lib\ifxjdbc.jar" -t informix_qx_comments.properties -db %1 -host 192.168.1.20:29140 -port 29140 -u informix -p informix -server production_tcp -o %2 -connprops="CLIENT_LOCALE\\=en_US.utf8;DB_LOCALE\\=en_US.57372" -s informix
```
`informix_qx_comments.properties` should be located in `C:\Tools\SchemaSpy`

That should be it, I'm a bit unsure on the -connprops since they are weird. Something to change in v7
Only problem I could think of is if the qx_table_comments and qx_column_comments are in a different schema.

**Comment by begooden-it** [comment]
Hi,


sorry for the delay I have been caught by too many tasks...

I finally tested it on 6.0.0 rc2 and it works great as described. I now just have to write all the comments, but hey, what did you expect ;-) ?

REgarding the schema thing, what I can say is that the 'schema' functionality is not that much used at least by 'old' informix customers. Anyway, in informix, a table is always created  in a database, not in a schema. Until those tables are not implemented by HCL/IBM as system tables, this can work like this.

I have been looking at the progression of my RFE at IBM, which is slowly collecting votes, but collecting. Implementation might be Q3 next year if we push more. I can also talk with Informix engineering and tell them I have developed the functionality and it works.

In the meantime, maybe we we could adopt more 'informix standard' names for the tables, such as systabcomments and syscolcomments ? So you could insert the functionality in a future release.

What do you think?

Have a great week
Eric

**Comment by begooden-it** [comment]
Hi Nils,

and BTW Happy New year

Just wondering if the comments for Informix have been integrated in some newer version and if yes, what should be the syntax ?

I just posted another issue about Informix Check constraints not appearing.
https://github.com/schemaspy/schemaspy/issues/637

Thank you
Eric

**Comment by npetzall** [comment]
Hi, Thanks and double to you 2018->2019 and 2019->2020.

It has not been added to schemaspy, so you still need to use a custom databaseType (properties-file).
But if you have some good news about the RFE, this can be added to schemaspy.
